
# HG changeset patch
# User Timo Sirainen <tss@iki.fi>
# Date 1281449944 -3600
# Node ID 2281223b9cb99357a89971b64680530813d22a81
# Parent  367ce71922bfc95ec1faad91cd47f1c758cd2d50
master: Don't crash on config reload when using dict processes.

--- a/src/master/dict-process.c	Wed Aug 04 16:54:17 2010 +0100
+++ b/src/master/dict-process.c	Tue Aug 10 15:19:04 2010 +0100
@@ -214,7 +214,8 @@
 	struct dict_listener *listener = process->listener;
 
 	dict_process_deinit(process);
-	if (listener->processes == NULL && listener->fd != -1) {
+	if (listener->processes == NULL && listener->fd != -1 &&
+	    listener->io == NULL) {
 		/* last listener died, create new ones */
 		listener->io = io_add(listener->fd, IO_READ,
 				      dict_listener_input, listener);

